doc/neps/nep-0021-advanced-indexing.rst

```diff
@@ -218,36 +329,28 @@ confusing.
 
 To facilitate implementations we could provide functions similar to
 ``operator.itemgetter`` and ``operator.setitem`` for the attributes.
-Possibly a MixIn could be provided to help implementation, but further
-ideas on this are necessary.
-
-Related Operation
-~~~~~~~~~~~~~~~~~
-
-There exist a further indexing or indexing like method. That is the
-inverse of a command such as ``np.argmin(arr, axis=axis)``, to pick
-the specific elements *along* an axis given an array of (at least
-typically) the same size.
-
-These function are added to NumPy in versoin 15 as `take_along_axis` and
-`put_along_axis`.
-
+Possibly a mixin could be provided to help implementation. These improvements
+are not essential to the initial implementation, so they are saved for
+future work.
```

Comment:
I do not disagree, but I have a bit of reservations.
 1. We have to figure out subclasses internally, mostly for masked arrays (I do not care about matrix really, it is horribly broken, and maybe OK to not to support them. The mmap subclass is pretty simple so also not a big deal).
 2. I am still not quite sure if ideas such as `__numpy_getitem__` (or a kwarg to getitem itself) may be the best way.

Or should we just ignore it for MA. Probably I am just not doing enough with funny array-likes to have a quite clear plan.

Reply:
I think it would be fine not to immediately implement oindex/vindex for masked arrays, as long as we don’t break existing behavior. Somebody who cares about masked arrays will need to work on it.

Reply:
> or a kwarg to getitem itself

Other alternatives I can think of for the meaning of `arr.oindex[a, b, c]`:

* `arr[np.core.orthogonal_index(a, b, c)]` - will work fine by default on maskedarrays, and any subclasses that just forward the indexer to the base class
* `arr[np.core.orthogonal_indexing, a, b, c]` (like C++ tag-dipatching)

Reply:
Well, one thing to consider is, that if MA does not support it and we implement DeprecationWarnings, for example matplotlib might have no way to go forward with the warnings? Though thta might be a general problem, since backward compatibility and avoiding warnings is tricky. It would be good to find a solution for that!

Reply:
it would probably be quite easy to make masked array’s ‘__getitem__‘ use ‘legacy_index‘ internally. That’s about as far as I would be willing to go, personally.

Reply:
Fair point, though it feels a bit half arsed to me :). With a bit of effort from someone not me we could have had that two years ago, hehe...